ddtrace/opentelemetry/_trace.py

```diff
@@ -89,9 +88,8 @@ def start_span(
             # Otel span was not generated by the ddtrace library and does not have an underlying ddtrace span.
             # Convert otel span to a ddtrace context object.
             trace_id, span_id, _, tf, ts, _ = curr_otel_span.get_span_context()
-            sp = 1 if tf is TraceFlags.SAMPLED else 0
-            meta = {W3C_TRACESTATE_KEY: ts.to_header()} if ts else None  # type: Optional[_MetaDictType]
-            dd_active = ddtrace.context.Context(trace_id, span_id, sampling_priority=sp, meta=meta)
+            trace_state = ts.to_header() if ts else None
+            dd_active = _TraceContext._get_context(trace_id, span_id, tf, trace_state)
```

Comment:
TraceFlag might be `00` (do not sample) but TraceState might have `dd=s:2` (manual keep). Discrepancies between sampling decisions in traceflags and tracestate should be resolved by `_TraceContext`.